### ISSUE TITLE
Add an option for a tree hierarchy view

### ISSUE THREAD

**Issue by PaBox** [Kunzisoft/KeePassDX#958]
**Is your feature request related to a problem? Please describe.**
It is a hazzle to view multiple entries in different groups and since the first time I of starting KeePassDX (coming from KeePassXC) I am getting confused with the arrow on the left side of each Group, which serves no functional purpose, other then to visualize a group entry.

**Describe the solution you'd like**
It would be good to implement an option to trigger a different view mode, which allows for the visualization of subgroups and -entries. The arrow would rotate and show the entries underneath, while the inner view for all entries must be configured to be able to slide left and optionally snap to the right side of the last opened entry. 

**Describe alternatives you've considered**
The view must not snap, if inconvinient. But I do not think of an alternative, as the only other efficient solution is implemented atm. Showing the subentries in a temporary window would not allow for a recursive hierarchy and nothing else came to my head. This solution is also implemented in probably any desktop-file-browser and also KeePassXC.

**Additional context**
It needs time and resources to find multiple entries with unknowm names, which is a hazzle for any hardcore-user. It would benefit a lot and help with headpain when using the app to often. People actually work on their phones so why not make it easier ;)

**Comment by braoult** [comment]
I like it ! I currently have to go back to my desktop and use `KeePassXC` when I need to fix a `syncthing`collision.

To do that, I open the 2 databases, select all entries (this is a table view with any field I need), and sort by modification time.
It becomes then possible, by checking that modification time field in displayed tables, to find out the "forking" starting point.

**Comment by J-Jamet** [comment]
The file collisions are not really related to the functionality of displaying the hierarchical view.
@braoult For your problem, it is much easier to use the merge feature, available in KeePassDX and KeePassXC.

But I agree, the hierarchical view would be a good addition.

**Comment by braoult** [comment]
Exact, I was more thinking about a table view than pure hierarchical view.

About merge (at least on KeepassXC), I prefer to avoid it until we can make a "diff" first ;-)
